### PR TITLE
App schema solr Integration test

### PR DESCRIPTION
**Pull request by fernandor777**

### CODE REVIEW

File: modules/extension/app-schema/app-schema/pom.xml [hunk 1]
```diff
@@ -188,6 +188,18 @@
...
+            <artifactId>gt-solr</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+		      <groupId>org.slf4j</groupId>
+		      <artifactId>slf4j-api</artifactId>
+		      <version>1.7.7</version>
+		      <scope>test</scope>
+		    </dependency>
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/pom.xml]: The Apache Solr integration tests should be moved to ``gt-solr`` module.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+import org.opengis.feature.Feature;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+/**
+ * app-schema - Apache Solr Integration Test
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: I would put something like: ``This class contains the integration tests (online tests) for the integration between App-Schema and Apache Solr``.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+ *
+ * @author Fernando Miño
+ */
+public class SolrIntegrationTest extends OnlineTestCase {
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: Id this class is not to be extended mark it as ``final``.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+    protected static final String SOLR_URL_KEY = "solr_url";
+    protected static final String CORE_NAME = "stations";
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: This two need to be protected ?

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+    private static final String testData = "/test-data/solr/";
+    private static final String xmlFileName = "mappings_solr.xml";
+    private static final String NSURI = "http://www.stations.org/1.0";
+    // private static final Name mappedTypeName = Types.typeName(NSURI, "multi_stations_solr");
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: You probably want to remove this :tongue:.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+            "target/test/" + SolrIntegrationTest.class.getSimpleName();
+    private static final File testDir = new File(testDirStr);
+
+    private static final File appSchemaCacheDir =
+            new File(
+                    "target/test/"
+                            + SolrIntegrationTest.class.getSimpleName()
+                            + "/app-schema-cache");
+
+    protected HttpSolrClient client;
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: Does it need to be ``protected``.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+    protected void createField(String name, String type, boolean multiValued) {
+        TestsSolrUtils.createField(client, name, type, multiValued);
+    }
+
+    protected void prepareFiles() throws Exception {
+        // copy meteo.xsd
+        copyTestData("meteo.xsd", testDir);
+
+        // Modify datasource and copy xml
+        File xmlFile =
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: This is fine, but another alternative could be putting some placeholder in the file like ``${PLACE_HOLDER}`` and then substitute with a replace all, more easy and cleaner.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/util/TestsSolrUtils.java [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * This class contains helper methods typically used when dealing with Apache Solr integration
+ * tests.
+ */
+public final class TestsSolrUtils {
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/util/TestsSolrUtils.java]: As discussed this duplication should be avoided.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+    @Test
+    public void testFeaturesCount() throws Exception {
+        FeatureSource<FeatureType, Feature> fSource =
+                (FeatureSource<FeatureType, Feature>)
+                        mappingDataStore.getFeatureSource(
+                                new NameImpl("http://www.stations.org/1.0", "Station"));
+        assertEquals(7, size(fSource.getFeatures()));
+    }
+
+    private int size(FeatureCollection<FeatureType, Feature> features) {
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: This should also check that the expected features are present.

File: modules/extension/app-schema/app-schema/pom.xml (hunk 1 above)
Comment (fernandor777) [reply modules/extension/app-schema/app-schema/pom.xml]: Done

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java (hunk 2 above)
Comment (fernandor777) [reply modules/extension/app-schema/app-schema/src/test/java/org/geotools/test/SolrIntegrationTest.java]: Done, instructions added.

File: modules/extension/app-schema/app-schema/src/test/resources/test-data/solr/mappings_solr.xml [hunk 10]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/resources/test-data/solr/mappings_solr.xml]: This file should have been removed right ? or better moved to the solr module ?

File: modules/extension/app-schema/app-schema/src/test/resources/test-data/solr/meteo.xsd [hunk 11]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0"?>
```
Comment (nmco) [inline modules/extension/app-schema/app-schema/src/test/resources/test-data/solr/meteo.xsd]: Same here.

File: modules/unsupported/solr/src/test/java/org/geotools/data/solr/AppSchemaIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+
+    /** solr.properties file required */
+    @Override
+    protected String getFixtureId() {
+        return "appschema";
+    }
+
+    /** Station data model class */
+    @XmlRootElement
+    public static class StationData implements Serializable {
```
Comment (nmco) [inline modules/unsupported/solr/src/test/java/org/geotools/data/solr/AppSchemaIntegrationTest.java]: Do we still need this ? For reading the file cotnaining the data ?

**Comment by nmco** [comment]
Looks like there is a few files that where no deleted. The code structure coudl be improved, e.g. long function class too bigs. But htat's something we can do later.

**Pull request merged**